src/MIPSolver/MIPSolverCbc.cpp

```diff
@@ -31,6 +32,44 @@
 namespace SHOT
 {
 
+class TerminationEventHandler : public CbcEventHandler, public MIPSolverCallbackBase
+{
+
+public:
+    TerminationEventHandler(EnvironmentPtr envPtr) { env = envPtr; };
+    // EventHandler(CbcModel* model) : CbcEventHandler(model) {};
+
+    virtual ~TerminationEventHandler() {};
+
+    TerminationEventHandler(const TerminationEventHandler& rhs) : CbcEventHandler(rhs) {};
+
+    TerminationEventHandler& operator=(const TerminationEventHandler& rhs)
+    {
+        if(this != &rhs)
+        {
+            CbcEventHandler::operator=(rhs);
+        }
+        return *this;
+    }
+
+    virtual TerminationEventHandler* clone() const { return new TerminationEventHandler(env); };
+
+    virtual CbcAction event(CbcEvent whichEvent)
+    {
+        if(whichEvent == CbcEventHandler::CbcEvent::node)
+        {
+            env->output->outputDebug("        Terminated by user.");
+
+            // return (CbcEventHandler::CbcAction::stop);
```

Comment:
Shouldn't it check with SHOT here whether it should return `stop` ?

Reply:
> Shouldn't it check with SHOT here whether it should return `stop` ?

Yes, it should :)